src/extension/wps/src/main/java/au/org/emii/ncdfgenerator/NcdfEncoder.java

```diff
@@ -290,60 +291,70 @@ private void populateValues(
+                    dimensionsMap.put(dimension.getName(), dimension);
+                }
 
-        // pre-map the encoders by index according to the column name
-        ArrayList<IAddValue>[] processing = (ArrayList<IAddValue>[]) new ArrayList[numColumns + 1];
+                // organize variables by name
+                Map<String, IVariable> variablesMap = new HashMap<String, IVariable>();
+                for (IVariable variable : definition.getVariables()) {
+                    variablesMap.put(variable.getName(), variable);
+                }
 
-        for (int i = 1; i <= numColumns; ++i) {
+                // pre-map the encoders by index according to the column name
+                ArrayList<IAddValue>[] processing = (ArrayList<IAddValue>[]) new ArrayList[numColumns + 1];
 
-            processing[i] = new ArrayList<IAddValue>();
+                for (int i = 1; i <= numColumns; ++i) {
 
-            IDimension dimension = dimensionsMap.get(m.getColumnName(i));
-            if (dimension != null) {
-                processing[i].add(dimension);
-            }
+                    processing[i] = new ArrayList<IAddValue>();
 
-            IAddValue variable = variablesMap.get(m.getColumnName(i));
-            if (variable != null) {
-                processing[i].add(variable);
-            }
-        }
+                    IDimension dimension = dimensionsMap.get(m.getColumnName(i));
+                    if (dimension != null) {
+                        processing[i].add(dimension);
+                    }
 
-        // process result set rows
-        while (rs.next()) {
-            for (int i = 1; i <= numColumns; ++i) {
-                for (IAddValue p : processing[i]) {
-                    p.addValueToBuffer(rs.getObject(i));
+                    IAddValue variable = variablesMap.get(m.getColumnName(i));
+                    if (variable != null) {
+                        processing[i].add(variable);
+                    }
+                }
+
+                // process result set rows
+                while (rs.next()) {
+                    for (int i = 1; i <= numColumns; ++i) {
+                        for (IAddValue p : processing[i]) {
+                            p.addValueToBuffer(rs.getObject(i));
+                        }
+                    }
                 }
             }
-        }
+       }
     }
 
-    public void closeQuietly() {
+    public void close() {
+        closeQuietly(featureInstancesRS);
+        closeQuietly(featuresStmt);
+        closeQuietly(conn);
+    }
+
+    private void closeQuietly(AutoCloseable autoCloseable) {
+        if (autoCloseable == null) return;
+
         try {
-            conn.close();
-        }
-        catch (SQLException e) {
-            logger.info("problem closing transaction");
+            autoCloseable.close();
+        } catch (Exception e) {
+            logger.warn(String.format("Problem closing %s", autoCloseable.getClass().getCanonicalName()), e);
```

Comment:
What is the result of evaluating, 

```
String.format("Problem closing %s", autoCloseable.getClass().getCanonicalName())
```

Isn't it just 'AutoCloseable'

Is the actual exception given by 'e' reported in the logger.warn statement?